### DISCUSSION TITLE
Is there a package that provides an xarray-based implementation of `cdutil.vertical.reconstructPressureFromHybrid`?

### DISCUSSION THREAD

**Discussion by tomvothecoder** [xCDAT/xcdat#440] [2. Questions]
Hi @xCDAT/core-developers or anybody in the xCDAT community. Have any of you come across a package that implements an xarray-based function similar to [`cdutil.vertical.reconstructPressureFromHybrid`](https://github.com/CDAT/cdutil/blob/b823b69db46bb76536db7d435e72075fc3975c65/cdutil/vertical.py#L8-L49)? I am working on refactoring `e3sm_diags` and `e3sm_to_cmip` to use xarray/xCDAT and there are references to this function that I need to replace. Thanks!

Docstring:
```
def reconstructPressureFromHybrid(ps, A, B, Po):
    """
    Reconstruct the Pressure field on sigma levels, from the surface pressure
    :param Ps: Surface pressure
    :param A: Hybrid Conversion Coefficient, such as: p=B.ps+A.Po.
    :param B: Hybrid Conversion Coefficient, such as: p=B.ps+A.Po.
    :param Po: Hybrid Conversion Coefficient, such as: p=B.ps+A.Po
    :param Ps: surface pressure
    .. note::
        A and B are 1d sigma levels.
        Po and Ps must have same units.
    :returns: Pressure field, such as P=B*Ps+A*Po.
    :Example:
        .. doctest:: vertical_reconstructPressureFromHybrid
            >>> P=reconstructPressureFromHybrid(ps,A,B,Po)
    """
	...
```

Examples Usages:
 - https://github.com/E3SM-Project/e3sm_diags/blob/4a82b0ec3b2a81233b294fa505c6aa8228ecbd27/e3sm_diags/driver/utils/general.py#L132-L146
 - https://github.com/E3SM-Project/e3sm_to_cmip/blob/70f7dad381ef304b022826b540df0e7b46ba9912/e3sm_to_cmip/cmor_handlers/vars/pfull.py#L38-L47
 - https://github.com/E3SM-Project/e3sm_to_cmip/blob/70f7dad381ef304b022826b540df0e7b46ba9912/e3sm_to_cmip/cmor_handlers/vars/phalf.py#L39-L50

**Comment by chengzhuzhang** [comment]
Hi Tom, 
We had an example in the vertical interpolation PR: Line 14 and forward in https://github.com/xCDAT/xcdat/blob/9e2800ccb77f6f118f74429ab40052019206e5d6/docs/examples/regridding-vertical.ipynb

We can wrap it in a function to construct pressure based on hybrid parameters, here is an example how geocat codes it https://github.com/NCAR/geocat-comp/blob/0263d2d77ab22a6be4d1d3f16d5bfcfc68057364/src/geocat/comp/interp_hybrid_to_pressure.py#L77

**Reply by tomvothecoder** [reply]
Thanks Jill!

**Comment by jasonb5** [comment]
`cf-xarray` doesn't currently support reconstructing pressure from hybrid but it provides a framework for this via `decode_vertical_coords` method: https://cf-xarray.readthedocs.io/en/latest/parametricz.html.

Might be an opportunity to make a PR and fill out support for decoding other [parametric vertical coordinates](https://cfconventions.org/cf-conventions/cf-conventions.html#_atmosphere_hybrid_sigma_pressure_coordinate).
